internal/data_test.go

```diff
@@ -69,21 +70,44 @@ func TestConvertData(t *testing.T) {
 		e       interface{} // Expected result.
 	}{
 		{"bool", ddl.Bool{}, false, "", "true", true},
-		{"bytes", ddl.Bytes{Len: ddl.MaxLength{}}, false, "", `\\x0001beef`, []byte{0x0, 0x1, 0xbe, 0xef}},
+		{"bytes", ddl.Bytes{Len: ddl.MaxLength{}}, false, "", `\x0001beef`, []byte{0x0, 0x1, 0xbe, 0xef}},
```

Comment:
Should we also change the return value of scalar types to spanner.Null*? Because now they are inconsistent, e.g., convert a single bool value (bool type in Go) and an array of bool values (an array of spanner.NullBool).

Reply:
So for scalars we already handle the case of NULL in ConvertData: NULL columns are represented by pgdump using "\\n" which we handle in the main for loop (line 89). Basically the for loop builds up both a list of columns and values, and simply skips columns if the data is NULL.

Array, on the other hand, are different: if you have an array value that is NULL, you can't just skip it!

Having said that, I might need to revisit this when dealing with data returned from select statements. But let's leave leave as is for now.